test/k8s/simple_test.go

```diff
@@ -146,34 +150,52 @@ spec:
 	})
 
 	Context("scaling", func() {
+		var admin envoy_admin.Tunnel
+
+		BeforeAll(func() {
+			pod := k8s.ListPods(
+				cluster.GetTesting(),
+				cluster.GetKubectlOptions(TestNamespace),
+				metav1.ListOptions{
+					LabelSelector: "app=fake-client",
+				},
+			)[0]
+			tnl := k8s.NewTunnel(cluster.GetKubectlOptions(TestNamespace), k8s.ResourceTypePod, pod.Name, 0, 9901)
+			Expect(tnl.ForwardPortE(cluster.GetTesting())).To(Succeed())
+			admin = tunnel.NewK8sEnvoyAdminTunnel(cluster.GetTesting(), tnl.Endpoint())
+		})
+
+		srv := "srv-000"
 		scale := func(replicas int) {
 			err := k8s.RunKubectlE(
 				cluster.GetTesting(),
 				cluster.GetKubectlOptions(TestNamespace),
-				"scale", "statefulset", "srv-000", fmt.Sprintf("--replicas=%d", replicas),
+				"scale", "statefulset", srv, fmt.Sprintf("--replicas=%d", replicas),
 			)
 			Expect(err).ToNot(HaveOccurred())
 
-			err = cluster.Install(WaitNumPods(TestNamespace, replicas, "srv-000"))
+			err = cluster.Install(WaitNumPods(TestNamespace, replicas, srv))
 			Expect(err).ToNot(HaveOccurred())
+			start := time.Now()
+			Eventually(func(g Gomega) {
+				membership, err := admin.GetStats(fmt.Sprintf("cluster.%s_kuma-test_svc_80.membership_total", srv))
```

Comment:
We can take envoy_cluster_membership_total from Prometheus, but we have to enable metrics on the Mesh resource. TBH I think it'd be easier to use the same approach for all assertions rather than use Envoy Admin API in this particular case

Reply:
That's what I had previously but @jakubdyszkiewicz objected. Note that metrics are already enabled as is because we set `WithMesh: true`.

Reply:
The other metrics we query from prom are for the CP, so we can't use admin

Reply:
@jakubdyszkiewicz what are the advantages to use Envoy Admin API rather than Prometheus?

Reply:
> The other metrics we query from prom are for the CP, so we can't use admin

Yes, I meant to use prom for everything

Reply:
yeah I just wanted to note that I don't think it's necessarily a bad thing to use separate approaches because CP and DPs are really separate pieces

Reply:
The objection was around whether collecting DP metrics scales once we're at thousands of Dataplanes.

Reply:
I think we can disable them being collected or scraped.

Reply:
But it's good to have them, no? We advise users to enable Envoy metrics collection on the Mesh, if it doesn't scale well then maybe we're exposing too many metrics

Reply:
I don't think we should enable metrics from DPPs at this moment. This effort will very quickly become "how to scale prometheus to handle X DPPs". Especially if we want to scrape every 1s. I'd expect the artifact with prom snapshot to be very large with 10k Pods.

Reply:
I guess [we enable metrics from DPPs at the moment](https://github.com/kumahq/kuma-tools/blob/c9a6c29d9b460863ef1fa1c964acd26f59229823/graph/generate_mesh.go#L133) but they aren't [successfully scraped by prometheus because of mTLS](https://github.com/kumahq/kuma-tools/blob/c9a6c29d9b460863ef1fa1c964acd26f59229823/graph/generate_mesh.go#L124)